qiita_db/metadata_template/prep_template.py

```diff
@@ -989,7 +999,7 @@ def _get_predecessors(workflow, node):
                     init_artifacts = {
                         wkartifact_type: f'{starting_job.id}:'}
                 else:
-                    init_artifacts = {wkartifact_type: self.artifact.id}
+                    init_artifacts = {wkartifact_type: str(self.artifact.id)}
```

Comment:
I hate to say it but having to cast an internal id to string is a little bit concerning to me. Internal code should use the numeric form of the artifact_id and it should also be easy to pass the numeric version for any REST call. May I ask why this is needed? I feel like even if it's needed, it would be much clearer to cast it right before it's used to make it obvious why it's needed. Sorry if that's not clear.